z2/src/docgen.rs

```diff
@@ -202,7 +273,9 @@ impl Docs {
         println!("After remove {0}", serde_yaml::to_string(&contents_map)?);
 
         while let Some(ref path_entry) = stack.pop() {
```

Comment:
Existing code but consider using https://docs.rs/walkdir/latest/walkdir/ to make your life easier.

Reply:
Yeah - I did consider it, but felt that for the amount of effort, I should avoid adding trivial dependencies (every dependency being a supply chain risk)